### ISSUE TITLE
$450 /month Heroku and still having memory issues

### ISSUE THREAD

**Issue by ACPK** [catarse/catarse#2779]
I have the app hosted with 2 2x Heroku Web dynos but have had hourly memory issues. Has anyone else hit this problem?

**Comment by ACPK** [comment]
The problem seems to be the ProjectsController#Index.

**Comment by vicnicius** [comment]
From our heroku metrics I see our memory usage pretty stable, with an average 2,815.2MB and a max 3,773.8MB.

**Comment by ACPK** [comment]
That's a huge amount. What dynos do you use as the performance dynos max at 2.5G and are $250/month?

**Comment by vicnicius** [comment]
@devton can tell you more on how our infrastructure is setup on heroku. We probably use a more robust performance dyno there, but I really don't know much about it. 

I do agree that it looks like we have a pretty large memory consumption there. But then again, I am talking outside my expertise zone here. For this @devton and also @adrianob could share more insights.

I took a look at this in the past to try to reduce the amount of resources we load on this page, as we now fetch postgrest to get the data there from client-side, so not only we are doing too much on the controller, but we are also doing it twice, if I am not mistaken. Huge room for improvement and definitely something we will tackle with the upcoming bigger epics. 

And I will be happy to review your PR!

**Comment by adrianob** [comment]
@ACPK we use 1 Performance-L dyno, which has 14GB of RAM for $500 a month. Keep in mind we do have a lot of users, but memory usage can certainly be improved.

**Comment by ACPK** [comment]
@adrianob - Thanks. Before I take a crack at it, is there any issues I'll hit / reasons why the ProjectsControll#index  is called using Rails rather than PostgREST?

**Comment by adrianob** [comment]
@ACPK I believe the index action in rails is only used for embedding project cards on external websites. It seems unlikely this is the cause of the memory problems, but I could be wrong.

**Comment by vicnicius** [comment]
yeah, I made a small confusion there. Was thinking of the [ProjectsController#show](https://github.com/catarse/catarse/blob/develop/app/controllers/projects_controller.rb#L168), not #index.

**Comment by adrianob** [comment]
@ACPK also rails is used to generate all meta tags, since google's crawler won't execute js last I checked.

**Comment by adrianob** [comment]
I was wrong about the project embedding. We mainly just render the meta tags and render some scripts etc. You should take a look at the ```catarse_bootstrap.html.slim``` view and see if there is anything that can be improved.

**Comment by ACPK** [comment]
@adrianob - But the meta tags would be one per page. It does seem like it's being generated by Rails.

**Comment by mitdrissia** [comment]
is the memory leak issue solved in the updated version @ACPK @adrianob @vicnicius 
How much users can it handle. For example if you have 1000 users is a 32gb ram core i7 enough to handle this load?
